source/common/http/conn_manager_config.h

```diff
@@ -439,10 +439,15 @@ class ConnectionManagerConfig {
   virtual bool shouldMergeSlashes() const PURE;
 
   /**
-   * @return if the HttpConnectionManager should remove the port from host/authority header
+   * @return if the HttpConnectionManager should remove the matching port from host/authority header
    */
   virtual bool shouldStripMatchingPort() const PURE;
```

Comment:
Even though we can't do it at the API level right now, perhaps make this a single function that returns an enum of the strip type? That would eventually work with a oneof and I think always strip is a superset anyway so it should be fine?